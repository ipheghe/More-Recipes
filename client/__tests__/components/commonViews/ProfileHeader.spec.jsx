import expect from 'expect';
import React from 'react';
import { mount, shallow } from 'enzyme';
import render from 'react-test-renderer';
import sinon from 'sinon';
import { HashRouter as Router } from 'react-router-dom';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import ConnectedUserProfileHeader, { PureProfileHeader }
  from '../../../src/commonViews/ProfileHeader';


const middleware = [thunk];
const mockStore = configureMockStore(middleware);
const initialState = {};

const store = mockStore(initialState);

const state = {
  keyword: '',
};

const props = {
  getRecipesBySearch: jest.fn(() => Promise.resolve())
};

const event = {
  preventDefault: jest.fn(),
  target: {
    keyword: 'spaghetti'
  }
};

/**
 * @description setup function to mount component
 *
 * @return { object } mountedWrapper, shallowWrapper
 */
const setup = () => {
  const mountedWrapper = mount(<Router>
    <ConnectedUserProfileHeader {...props} store={store} />
  </Router>);
  const shallowWrapper = shallow(<PureProfileHeader {...props} />);
  return {
    mountedWrapper,
    shallowWrapper
  };
};

describe('<ProfileHeader', () => {
  it('renders ProfileHeader component without crashing ', () => {
    const { mountedWrapper } = setup();
    mountedWrapper.setState(state);
    expect(mountedWrapper).toBeDefined();
    expect(mountedWrapper.find('Link').length).toBe(1);
    expect(mountedWrapper.find('nav').length).toBe(1);
    expect(mountedWrapper.find('input').length).toBe(1);
    expect(mountedWrapper.exists()).toBe(true);
  });

  it('should match component snapshot', () => {
    const tree = render.create(<Router >
      <PureProfileHeader {...props} />
    </Router>);
    expect(tree).toMatchSnapshot();
  });

  it('calls handleChange event', () => {
    sinon.spy(PureProfileHeader.prototype, 'handleChange');
    const { shallowWrapper } = setup();
    shallowWrapper.setState(state);
    shallowWrapper.instance().handleChange(event);
    expect(PureProfileHeader.prototype.handleChange.calledOnce).toEqual(true);
  });

  it(`dispatches getRecipesBySearch action
    when search button is clicked`, () => {
      sinon.spy(PureProfileHeader.prototype, 'handleSearch');
      const { shallowWrapper } = setup();
      shallowWrapper.instance().handleSearch(event);
      expect(PureProfileHeader.prototype.handleSearch.calledOnce).toEqual(true);
    });
});
