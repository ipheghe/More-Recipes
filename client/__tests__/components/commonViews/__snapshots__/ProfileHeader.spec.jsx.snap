// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileHeader should match component snapshot 1`] = `
<nav
  className="navbar navbar-toggleable-md navbar-light bg-faded"
>
  <button
    aria-controls="navbarTogglerDemo01"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarTogglerDemo01"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarTogglerDemo01"
  >
    <a
      className="navbar-brand invisible-button"
      href="#/dashboard/top-recipes"
      onClick={[Function]}
    >
      <h2>
        Profile Page
      </h2>
    </a>
    <ul
      className="navbar-nav mr-auto mt-2 mt-lg-0"
    >
      <li
        className="nav-item"
      >
        <button
          className="nav-link invisible-button disabled"
        />
      </li>
    </ul>
    <form
      className="form-inline my-2 my-lg-0"
    >
      <input
        className="form-control mr-sm-2"
        formNoValidate={true}
        name="keyword"
        onChange={[Function]}
        placeholder="Search"
        required={true}
        type="text"
        value=""
      />
      <button
        className="btn btn-outline-success my-2 my-sm-0"
        id="search-recipe"
        onClick={[Function]}
      >
        Search
      </button>
    </form>
  </div>
</nav>
`;
